scripts/anvil_tools/set_up_anvil_workspaces.py

```diff
@@ -0,0 +1,255 @@
+    # -H  "accept: application/json" -H  "Authorization: Bearer [token]"
+    headers = {"Authorization": "Bearer " + get_access_token(), "accept": "application/json"}
+
+    # capture response from API and parse out status code
+    response = requests.post(uri, headers=headers)
+    status_code = response.status_code
+
+    # print success or fail message based on status code
+    if status_code == 201:
+        print(f"Successfully created Authorization Domain with name: {auth_domain_name}.")
+        return status_code
+    elif status_code in ["403", "409"]:
+        print(f"WARNING: Failed to create Authorization Domain with name: {auth_domain_name}.")
+        return "Authorization Domain with name already exists. Try again with unique group name."
+    else:
+        print(f"WARNING: Failed to create Authorization Domain with name: {auth_domain_name}.")
+        print("Check output file for error details.")
+        return(response.json())
+
+
+def setup_workspaces(tsv, project="anvil-datastorage"):
+    """Get the workspace and associated auth domain from input tsv file."""
+
+    # read full tsv into dataframe
+    setup_info_df = pd.read_csv(tsv, sep="\t")
+
+    # create df for output tsv file
+    col_names = ["input_workspace_name", "input_auth_domain_name",
+                 "auth_domain_email", "auth_domain_creation_error",
+                 "workspace_link", "workspace_creation_error",
+                 "workspace_ACLs", "workspace_ACLs_error",
+                 "workspace_setup_status"]
+    output_df = pd.DataFrame(columns=col_names)
+
+    # per row in tsv/df
+    for row in setup_info_df.index:
+        # authorization domain
+        auth_domain_name = setup_info_df.loc[row].get(key='auth_domain_name')
+        output_df.at[row, "input_auth_domain_name"] = auth_domain_name
+        auth_status = create_auth_domain(auth_domain_name)
+
+        if auth_status != 201:  # if auth creation fail
+            output_df.at[row, "auth_domain_creation_error"] = auth_status  # write error response to col
+            continue  # go to next row in df
+        else:  # if auth creation success
+            # TODO: will i need auth_domain_email again?
+            auth_domain_email = f"{auth_domain_name}@firecloud.org"
+            output_df.at[row, "auth_domain_email"] = auth_domain_email
+
+        # workspace
+        workspace_name = setup_info_df.loc[row].get(key='workspace_name')
+        output_df.at[row, "input_workspace_name"] = workspace_name
+        create_ws_request = make_create_workspace_request(workspace_name, auth_domain_name, project)  # get json for createWorkspace
+
+        create_workspace_status = create_workspace(create_ws_request, workspace_name, project)  # createWorkspace
+
+        # created or user wants to continue with existing
+        if create_workspace_status in [201, "Y"]:
+            output_df.at[row, "workspace_link"] = f"https://app.terra.bio/#workspaces/{project}/{workspace_name}"
```

Comment:
note that if `workspace_name` has spaces in it, you need to replace each space with `%20` for the url